### DISCUSSION TITLE
Reuse cross-container to use Maturin

### DISCUSSION THREAD

**Discussion by RaulTrombin** [cross-rs/cross#1267] [Q&A]
Someone can give a light here?
I'm trying to discover what cross command does:

When I do a:
cross build --target armv7-unknown-linux-gnueabihf
It opens a container,
 (with docker ps)
IMAGE http://ghcr.io/cross-rs/armv7-unknown-linux-gnueabihf:0.2.5 
COMMAND "sh -c 'PATH=$PATH:/rust/bin **cargo** build --target armv7-unknown-linux-gnueabihf'" 
I would like to know how to call this container from my terminal,

Basically what I would like is to install maturin on it.
```
FROM ghcr.io/cross-rs/armv7-unknown-linux-gnueabihf
RUN apt update && apt install -y --no-install-recommends python3 python3-pip && pip3 install -U pip
RUN pip3 install maturin
```
Then instead of doing what cross does, call **maturin** and not cargo.

**Comment by Emilgardis** [comment]
If you add `-v` you'll see what happens

I've outlined the steps in [this reddit comment](https://www.reddit.com/r/rust/comments/11wiwfu/how_does_cargo_cross_work/jd2cp2h/)

this seems related to https://github.com/cross-rs/cross/issues/1266 btw
